lib/alpha_beta.py

```diff
@@ -92,37 +103,29 @@ def check_conflict(current_cell: ndarray, player_index: int):
             cell[player_index] += nb_humans
         else:
             probability_of_win = nb_player / (2 * nb_humans)
-            battle_won = binomial(1, probability_of_win)
-            if battle_won:
-                added_humans = binomial(nb_humans, probability_of_win)
-                cell[0] = 0
-                cell[player_index] = (
-                    binomial(nb_player, probability_of_win) + added_humans
-                )
-            else:
-                cell[player_index] = 0
-                cell[0] = binomial(nb_humans, 1 - probability_of_win)
+            cell[player_index] = 0
+            cell[0] = binomial(nb_humans, 1 - probability_of_win)
     elif nb_opponent > 0 and nb_player > 0:
         if nb_player >= 1.5 * nb_opponent:
             cell[opponent_index] = 0
         else:
-            if nb_player >= nb_opponent:
+            if nb_player > nb_opponent:
                 probability_of_win = nb_player / nb_opponent - 0.5
-            else:
-                probability_of_win = nb_player / (2 * nb_opponent)
-            battle_won = binomial(1, probability_of_win)
-            if battle_won:
                 cell[opponent_index] = 0
-                cell[player_index] = binomial(nb_player, probability_of_win)
+                cell[player_index] = round(probability_of_win * nb_player)
             else:
+                probability_of_win = nb_player / (2 * nb_opponent)
                 cell[player_index] = 0
-                cell[opponent_index] = binomial(nb_opponent, 1 - probability_of_win)
+                cell[opponent_index] = round((1 - probability_of_win) * nb_player)
     return cell
```

Comment:
Could be nice to add a comment to explain we're implementing a workaround here because we want the AI to be cautious and avoid random battles.